sources/editor/Stride.Assets.Presentation/AssetEditors/ProjectWatcher.cs

```diff
@@ -86,23 +84,47 @@ public ProjectWatcher(SessionViewModel session, bool trackBinaries = true)
             fileChangedLink1 = fileChanged.LinkTo(fileChangedTransform);
             fileChangedLink2 = fileChangedTransform.LinkTo(AssemblyChangedBroadcast);
 
-            batchChangesTask = BatchChanges();
+            BatchChange = BatchChanges();
         }
 
-        private async Task BatchChanges()
+        private async IAsyncEnumerable<List<AssemblyChangedEvent>> BatchChanges()
         {
             var buffer = new BufferBlock<AssemblyChangedEvent>();
             using (AssemblyChangedBroadcast.LinkTo(buffer))
             {
-                while (true)
+                while (!batchChangesCancellationTokenSource.IsCancellationRequested)
                 {
                     var hasChanged = false;
                     var assemblyChanges = new List<AssemblyChangedEvent>();
                     do
                     {
                         var assemblyChange = await buffer.ReceiveAsync(batchChangesCancellationTokenSource.Token);
+
+
                         if (assemblyChange != null)
+                        {
                             assemblyChanges.Add(assemblyChange);
+                            var project = assemblyChange.Project;
+                            var projects = msbuildWorkspace.CurrentSolution.GetProjectDependencyGraph().GetProjectsThatTransitivelyDependOnThisProject(project.Id);
+                            foreach (var p in projects)
+                            {
+                                var assemblyName = msbuildWorkspace.CurrentSolution.GetProject(p).AssemblyName;
+                                var target = trackedAssemblies.FirstOrDefault(x => x.Project.AssemblyName == assemblyName);
+                                if (target != null)
+                                {
+                                    string file = assemblyChange.ChangedFile;
+                                    if (assemblyChange.ChangeType == AssemblyChangeType.Binary)
+                                    {
+                                        file = target.LoadedAssembly.Path;
+                                    }
+                                    else if (assemblyChange.ChangeType == AssemblyChangeType.Project)
+                                    {
+                                        file = target.Project.FilePath;
+                                    }
+                                    assemblyChanges.Add(new AssemblyChangedEvent(target.LoadedAssembly, assemblyChange.ChangeType, file, target.Project));
+                                }
+                            }
+                        }
 
                         if (assemblyChange != null && !hasChanged)
                         {
```

Comment:
How about swapping the first if here with 
```cs
if (assemblyChange == null) 
   continue;
```
And removing the first term of the second if, so just 
```cs
if (!hasChanged)
```

Reply:
did it